File: moldme/DTOs/ProjectDto.cs
using System.ComponentModel.DataAnnotations;
using moldme.Models;
namespace moldme.DTOs
{
    /// <summary>
    /// Data transfer object for Project model.
    /// </summary>
    public class ProjectDto
    {
        /// <summary>
        /// Project's unique identifier.
        /// </summary>
        [Required]
        [StringLength(64)]
        public string Name { get; set; }

        /// <summary>
        /// Project's unique identifier.
        /// </summary>
        [Required]
        [StringLength(256)]
        public string Description { get; set; }

        /// <summary>
        /// Project's unique identifier.
        /// </summary>
        [Required]
        public Status Status { get; set; }

        /// <summary>
        /// Project's unique identifier.
        /// </summary>
        [Required]
        [Range(0.01, double.MaxValue, ErrorMessage = "Budget must be greater than zero.")]
        public decimal Budget { get; set; }

        /// <summary>
        /// Project's unique identifier.
        /// </summary>
        [Required]
        public DateTime StartDate { get; set; }

        /// <summary>
        /// Project's unique identifier.
        /// </summary>
        [Required]
        public DateTime EndDate { get; set; }
    }
}

File: moldme_frontend/lib/models/project_dto.dart
class ProjectDto {
  final String name;
  final String description;
  final int status; // Integer to match dropdown value
  final double budget;
  final String startDate;
  final String endDate;

  ProjectDto({
    required this.name,
    required this.description,
    required this.status,
    required this.budget,
    required this.startDate,
    required this.endDate,
  });

  factory ProjectDto.fromJson(Map<String, dynamic> json) {
    return ProjectDto(
      name: json['name'],
      description: json['description'],
      status: json['status'], // Integer expected
      budget: json['budget'].toDouble(),
      startDate: json['startDate'],
      endDate: json['endDate'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'description': description,
      'status': status,
      'budget': budget,
      'startDate': startDate,
      'endDate': endDate,
    };
  }
}
